supportedBackends/sfs/src/main/scala/cromwell/backend/impl/sfs/config/ConfigHashingStrategy.scala

```diff
@@ -87,10 +96,73 @@ final case class HashPathModTimeStrategy(checkSiblingMd5: Boolean) extends Confi
   override val description = "hash file path and last modified time"
 }
 
-final case class HashFileStrategy(checkSiblingMd5: Boolean) extends ConfigHashingStrategy {
+final case class HashFileMd5Strategy(checkSiblingMd5: Boolean) extends ConfigHashingStrategy {
   override protected def hash(file: Path): Try[String] = {
     tryWithResource(() => file.newInputStream) { DigestUtils.md5Hex }
   }
 
-  override val description = "hash file content"
+  override val description = "hash file content with md5"
+}
+
+final case class HashFileXxH64Strategy(checkSiblingMd5: Boolean) extends ConfigHashingStrategy {
+  override protected def hash(file: Path): Try[String] = {
+    tryWithResource(() => file.newInputStream) {HashFileXxH64StrategyMethods.xxh64sum(_)}
+  }
+  override val description = "hash file content with xxh64"
+}
+
+final case class FingerprintStrategy(checkSiblingMd5: Boolean, fingerprintSize: Long) extends ConfigHashingStrategy {
+  override protected def hash(file: Path): Try[String] = {
+    Try {
+      // Calculate the xxh64 hash of last modified time and filesize. These are NOT added, as it will lead to loss of
+      // information. Instead their hexstrings are concatenated and then hashed.
+      HashFileXxH64StrategyMethods.xxh64sumString(file.lastModifiedTime.toEpochMilli.toHexString +
+      file.size.toHexString) +
+      HashFileXxH64StrategyMethods.xxh64sum(file.newInputStream, maxSize = fingerprintSize)
+      }
+    }
+  override val description = "fingerprint the file with last modified time, size and a xxh64 hash of the first 10 mb"
```

Comment:
description should be generalized for now-variable fingerprint size

Reply:
Now more general "first part of the file"